js/scripts.js

```diff
@@ -0,0 +1,202 @@
+    // set event listeners
+    builder.listenUp();
+    // set up input template
+    builder.addTemplate();
+}
+
+builder.getDOM = () => {
+    // dom elements
+    builder.resetButton = document.querySelector('.reset');
+    builder.buildButton = document.querySelector('.build');
+    builder.copyButton = document.querySelector('.clipboard');
+    builder.dlButton = document.querySelector('.download');
+    builder.input = document.querySelector('#userInput');
+    builder.output = document.querySelector('#builderOutput');
+}
+
+builder.listenUp = () => {
+    // event listeners
+    builder.resetButton.addEventListener('click', builder.reset);
+    builder.buildButton.addEventListener('click', builder.build);
+    builder.copyButton.addEventListener('click', builder.copy);
+    builder.dlButton.addEventListener('click', builder.download);
+}
+
+builder.addTemplate = () => {
+    // set template
+    builder.input.value = builder.template;
+}
+
+builder.reset = (e) => {
+    // prevent submit
+    e.preventDefault();
+    // call addTemplate method
+    builder.addTemplate();
+    // user confirmation
+    alert('Template has been reset for you!');
+}
+
+builder.build = (e) => {
+    // prevent submit
+    e.preventDefault();
+    // build json resume
+    // lacking data handling here for errors
+    builder.output.value = builder.input.value;
+    // user confirmation
+    alert('Successful build!');
+}
+
+//adapted from https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f
+builder.copy = (e) => {
+    // prevent submit
+    e.preventDefault();
+    // copy json resume, validate first if there's user output available
+    if (builder.output.value) {
+        builder.output.select();
+        document.execCommand('copy');
+        alert('Copied to clipboard!');
+    } else {
+        alert('Maybe you should build a JSON Resume first!');
+    }
+}
+
+builder.download = (e) => {
+    // prevent submit
+    e.preventDefault();
+    // download json resume only if there's user output available
+    if (builder.output.value) {
+        // convert string back into json object
+        const jsonObj = JSON.parse(builder.output.value);
+        // call download JSON method
+        builder.downloadJSON(jsonObj, 'resume');
+    } else {
+        alert('Maybe you should build a JSON Resume first!');
+    }
+}
+
+// adapted from https://stackoverflow.com/questions/19721439/download-json-object-as-a-file-from-browser
+builder.downloadJSON = (exportObj, exportName) => {
+    const dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(exportObj));
+    const downloadAnchorNode = document.createElement('a');
+    downloadAnchorNode.setAttribute("href",     dataStr);
+    downloadAnchorNode.setAttribute("download", exportName + ".json");
+    document.body.appendChild(downloadAnchorNode); // required for firefox
+    downloadAnchorNode.click();
+    downloadAnchorNode.remove();
+}
+
+builder.template = `{
```

Comment:
It might be worth considering how to handle invalid inputs filled into the template as a next step.